### ISSUE TITLE
Common modules between two packages elasticdl and elasticdl_client.

### ISSUE THREAD

**Issue by brightcoder01** [sql-machine-learning/elasticdl#1996]
We are planning to decouple elasticdl command line client from elasticdl main package. At the present, there are several python modules which are common shared between these two packages. We are listing these modules here:

```
elasticdl
--python
----common
------args.py
------constant.py.      (class: DistributionStrategy)
------k8s_client.py
------k8s_resource.py
------k8s_volumne.py
------model_utils.py.   (method: load_module)
```

We have two options to share these modules between these two packages.
1. Create a third package named `elasticdl_common`, put these modules into the packages. `elasticdl` and `elasticdl_client` have dependency on this package.
2. Keep these modules into the light-weight package `elasticdl_client`. The `elasticdl` main package has dependency on elasticdl_client.

My Proposal: There are not so many common modules, Option 2 is a more feasible choice.

**Comment by wangkuiyi** [comment]
I am not an expert in Python. I am not sure if it is possible to have three directories but two setup.py files.

```
elasticdl/
  server/
    setup.py
  common/
  client/
    setup.py
```

**Comment by brightcoder01** [comment]
> I am not an expert in Python. I am not sure if it is possible to have three directories but two setup.py files.
> 
> ```
> elasticdl/
>   server/
>     setup.py
>   common/
>   client/
>     setup.py
> ```

Just verified, we cannot add the python modules in `common` folder into `server` or `client` package using this structure above. 
Because the `setup.py` in server directory only packages all the files under the `server` folder. It cannot include the files in the `common` folder which is in the same level of `server` folder.

We can check all the options of directory structure in PR #2011 for discussion in detail.

**Comment by brightcoder01** [comment]
Discussion closed in PR #2011
